### PR TITLE
Can Message Rate Limiting

### LINKED ISSUE #127: Configurable Rate Limiting for CAN Messages
### Description

We want a way where we can set configurable rate limits for CAN messages. We want to set a minimum timeout before a CAN message of a certain type can be sent, and this timeout needs to be easily configurable. You should be able to go to somewhere like the bmsConfig file and change one number to change the rate limit of a message. Note: A lot of this should probably go in embedded base. It should be done in Shepherd first before it is made more abstract in embedded base.

### Acceptance Criteria

- [ ] Interface for rate limiting every CAN message
- [ ] Interface can be easily extended to accommodate new CAN messages or modifications to already existing messages
- [ ] If someone wants change the rate limit of a CAN message, they only have to change one number
- [ ] If someone wants to add a new CAN message, they do only have to change code in two locations in the codebase

### Proposed Solution

No clue

### PR DESCRIPTION
**Pull request by caiodasilva2005**
## Changes

added an interface to rate limit can messages

Closes #127

### CODE REVIEW

**Review by Sabramz — DISMISSED** [review]
your google doc

File: Core/Src/compute.c [hunk 1]
```diff
@@ -291,13 +358,11 @@ void compute_send_mc_discharge_message(acc_data_t *bmsdata)
 	endian_swap(&discharge_data.max_discharge,
 		    sizeof(discharge_data.max_discharge));
 
-	can_msg_t mc_msg = { 0 };
-	mc_msg.id =
-		0x156; // 0x0A is the dcl id, 0x22 is the device id set by us
-	mc_msg.len = 8;
-	memcpy(mc_msg.data, &discharge_data, sizeof(discharge_data));
+	bms_can_msgs[DISCHARGE].len = 8;
```
Comment (Sabramz) [inline Core/Src/compute.c]: The length of the message should be defined where the CAN message is defined now where it should be sent.

File: Core/Src/compute.c [hunk 2]
```diff
@@ -92,9 +92,76 @@ uint8_t compute_init(acc_data_t *bmsdata)
...
+{
+	can_msg_t discharge_msg = { 0 };
+	discharge_msg.id =
+		DISCHARGE_CANID; // 0x0A is the dcl id, 0x22 is the device id set by us
+
+	can_msg_t charge_msg = { 0 };
+	charge_msg.id =
+		CHARGE_CANID; // 0x0A is the dcl id, 0x157 is the device id set by us
+
+	can_msg_t acc_status_msg;
```
Comment (Sabramz) [inline Core/Src/compute.c:113]: You should put message lengths in the config since they do not change and are properties of the message.

**Review by Sabramz — CHANGES_REQUESTED** [review]

**Comment by jr1221** [comment]
This is gonna hella conflict with my CAN busy fixups, so you merge first and ill refactor it.

**Review by Sabramz — APPROVED** [review]
It works

**Pull request merged**